### PR TITLE
RavenDB-14089 Show encryption key and allow to change

### PR DESCRIPTION
**Pull request by Danielle9897**

### CODE REVIEW

File: src/Raven.Studio/typescript/models/database/tasks/periodicBackup/encryptionSettings.ts [hunk 1]
```diff
@@ -72,27 +84,62 @@ class encryptionSettings {
...
+            }
+        });
+        
+        this.key.subscribe(() => {
+            this.encryptionSection.syncQrCode();
+            this.keyConfirmation(false);
+        });
+
+        const dbName = ko.pureComputed(() => {
+            const db = activeDatabaseTracker.default.database();
```
Comment (ml054) [inline src/Raven.Studio/typescript/models/database/tasks/periodicBackup/encryptionSettings.ts]: it is ugly side effect, can we pass this from external ?

File: src/Raven.Studio/typescript/models/database/tasks/periodicBackup/encryptionSettings.ts [hunk 2]
```diff
@@ -158,22 +234,52 @@ class encryptionSettings {
...
+        });
 
-        const keyConfirmationNeeded = ko.pureComputed(() => this.canProvideOwnKey() && this.mode() === "UseProvidedKey");
+        this.changeKeyRequest.subscribe(changeKeyRequest => {
+            if (changeKeyRequest) {
+                return this.encryptionSection.generateEncryptionKey();
+            }
+        });
         
+        const keyConfirmationNeeded = ko.pureComputed(() => (this.canProvideOwnKey() && !this.originalKey()) ||
```
Comment (ml054) [inline src/Raven.Studio/typescript/models/database/tasks/periodicBackup/encryptionSettings.ts]: can we simply this expression?

File: src/Raven.Studio/typescript/models/database/tasks/periodicBackup/encryptionSettings.ts [hunk 3]
```diff
@@ -183,10 +289,22 @@ class encryptionSettings {
...
+    setEncryptionType(mode: Raven.Client.Documents.Operations.Backups.EncryptionMode) {
         this.mode(mode);
     }
 
+    setKeyUsedBeforeSave() {
+        // Use 'original key' for 'save' when: 
+        // opened existing encrypted backup and didn't request to change key
+        // or if 'key' does exist but asked to change 
+        // remember - 'key' has no value if wasn't generated from UI...
+        if ((!this.key() && !!this.originalKey()) ||
```
Comment (ml054) [inline src/Raven.Studio/typescript/models/database/tasks/periodicBackup/encryptionSettings.ts]: can we simply this? Also `!!` are not needed in javascript.

File: src/Raven.Server/Web/System/AdminServerWideBackupHandler.cs [hunk 4]
```diff
@@ -130,6 +131,9 @@ public Task GetServerWideBackupConfigurationCommand()
                 {
                     var backup = JsonDeserializationServer.ServerWideBackupConfiguration(backupBlittable);
                     backup.BackupDestinations = backup.GetDestinations();
+                    backup.IsEncrypted = backup.BackupEncryptionSettings != null &&
```
Comment (ppekrol) [inline src/Raven.Server/Web/System/AdminServerWideBackupHandler.cs:134]: Is this valid? It differs from the `IsBackupEncrypted` logic

Comment (grisha-kotler) [reply src/Raven.Server/Web/System/AdminServerWideBackupHandler.cs:134]: We only know that the configuration is encrypted

File: src/Raven.Studio/typescript/models/database/tasks/periodicBackup/encryptionSettings.ts (hunk 2 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/models/database/tasks/periodicBackup/encryptionSettings.ts]: changed to:
```
        const keyConfirmationNeeded = ko.pureComputed(() =>
             this.canProvideOwnKey() &&
             (!this.originalKey() || this.originalKey() && this.changeKeyRequest())
        );
```

Comment (ml054) [reply src/Raven.Studio/typescript/models/database/tasks/periodicBackup/encryptionSettings.ts]: could you put `(` and `)` around. Please never use `||` and `&&`  together without braces.

File: src/Raven.Studio/typescript/models/database/tasks/periodicBackup/encryptionSettings.ts (hunk 3 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/models/database/tasks/periodicBackup/encryptionSettings.ts]: changed to:

```
        if (this.originalKey() && 
           (!this.key() || (this.key() && !this.changeKeyRequest()))) {
            
            this.key(this.originalKey()); // using original
            this.keyConfirmation(true);   // needed for validation  
        }
```

File: src/Raven.Studio/typescript/models/database/tasks/periodicBackup/encryptionSettings.ts (hunk 2 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/models/database/tasks/periodicBackup/encryptionSettings.ts]: yes - done
```
        const keyConfirmationNeeded = ko.pureComputed(() =>
             this.canProvideOwnKey() &&
             (!this.originalKey() || (this.originalKey() && this.changeKeyRequest()))
        );
```

File: src/Raven.Studio/typescript/models/database/tasks/periodicBackup/encryptionSettings.ts (hunk 3 above)
Comment (ml054) [reply src/Raven.Studio/typescript/models/database/tasks/periodicBackup/encryptionSettings.ts]: !this.key() || !this.changeKeyRequest()

Comment (Danielle9897) [reply src/Raven.Studio/typescript/models/database/tasks/periodicBackup/encryptionSettings.ts]: done

File: src/Raven.Studio/typescript/models/database/tasks/periodicBackup/encryptionSettings.ts (hunk 2 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/models/database/tasks/periodicBackup/encryptionSettings.ts]: done

```
        const keyConfirmationNeeded = ko.pureComputed(() =>
            this.canProvideOwnKey() &&
            (!this.originalKey() || this.changeKeyRequest())
        );
```

File: src/Raven.Studio/typescript/models/database/tasks/periodicBackup/encryptionSettings.ts (hunk 1 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/models/database/tasks/periodicBackup/encryptionSettings.ts]: ok - done

**Review by ml054 — APPROVED** [review]

**Review by ppekrol — APPROVED** [review]

**Pull request merged**